.github/workflows/msi-build.yml

```diff
@@ -0,0 +1,55 @@
+name: msi-build
+
+on:
+  workflow_call:
+    inputs:
+      OS:
+        required: true
+        type: string
+
+env:
+  GO_VERSION: 1.23.10
+
+jobs:
+  msi-build:
+    runs-on: ubuntu-24.04
+    steps:
+      - name: Check out the codebase.
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
```

Comment:
Can we use this in other places also?

Reply:
I misread I thought that it was using fetch-depth 1, that in theory should be "cheaper", but, I guess that here we need all the repo history and tags.

Reply:
If there is a specific reason to not use the default here, i.e.: `fetch-depth: 1`, it will be good to have a comment

Reply:
I'm going to remove this in favor of default (as long as tests pass). This was directly copied without thinking about reasoning of fetch depth, I don't see a reason to go against default here, I don't see a need for all history since this is just building the most recent commit.